appinfo/Migrations/Version20170913113840.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * @author Sujith Haridasan <sharidasan@owncloud.com>
+ *
+ * @copyright Copyright (c) 2017, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Encryption\Migrations;
+
+use OCP\Migration\ISimpleMigration;
+use OCP\Migration\IOutput;
+
+class Version20170913113840 implements ISimpleMigration {
+
+	/**
+	 * @param IOutput $out
+	*/
+	public function run(IOutput $out) {
+		$installedVersion = \OC::$server->getConfig()->getSystemValue('version', '0.0.0');
+
+		if (version_compare('10.0.0', $installedVersion) !== 0) {
```

Comment:
Verified with the change by upgrading from 10.0.2 to 10.0.3. In both the user specific key or master key when enabled the later version doesn't show the dropdown list now.

Reply:
what you wrote here is basically `$version !== '10.0.0'` not `$version <= 10.0.3`

The reason it works is probably because the check in your "if" block would likely work correctly even without a version check. I'd rather have either the correct logic with version check, or no version check at all.

Furthermore if someone upgrades from 10.0.0 the logic below will not run at all.
Please recheck the documentation of `version_compare`